3-networks/modules/partner_interconnect/interconnect_attachment/main.tf

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_compute_interconnect_attachment" "attachment" {
```

Comment:
Do we need this as an inner module since it is just a single resource or can we directly just use this resource in partner_interconnect/main.tf

Reply:
It is matter of style. I would rather keep all common settings inside the module and pass parameters that I want to differ into it. 

I can move it to parent, but then the intent is not that clear. at least for me. module provides some logical boundary

Reply:
I agree but it seemed like `type = "PARTNER"` was the only common setting so I was wondering if we needed another layer.

Reply:
let me move it into the main TF code and see if I still like it :)